RUFAS/routines/manure/manure_manager.py

```diff
@@ -348,10 +344,158 @@ def request_nutrients(self, request: NutrientRequest) -> NutrientRequestResults:
 
         """
         if self.simulate_animals:
-            return self._manure_nutrient_manager.request_nutrients(request)
+            request_result, is_nutrient_request_fulfilled = self._manure_nutrient_manager.request_nutrients(request)
+            self._record_manure_request_results(request_result, "on_farm_manure")
+            if not is_nutrient_request_fulfilled and request.use_supplemental_manure:
+                self.om.add_log(
+                    "Supplemental manure needed",
+                    "Attempting to fulfill manure nutrient request shortfall with supplemental manure.",
+                    {"class": self.__class__.__name__, "function": self.request_nutrients.__name__},
+                )
+                amount_supplemental_manure_needed = self._calculate_supplemental_manure_needed(request_result, request)
+                supplemental_manure = self._field_manure_supplier.request_nutrients(amount_supplemental_manure_needed)
+                self._record_manure_request_results(supplemental_manure, "supplemental_manure")
```

Comment:
```suggestion
                self._record_manure_request_results(supplemental_manure, "off_farm_manure")
```
"off farm" might be more straightforward for users to understand.